junit-platform-suite-engine/src/main/java/org/junit/platform/suite/engine/DiscoverySelectorResolver.java

```diff
@@ -34,6 +33,7 @@ private static void discoverSuite(TestDescriptor descriptor) {
 
 	void resolveSelectors(EngineDiscoveryRequest request, SuiteEngineDescriptor engineDescriptor) {
 		resolver.resolve(request, engineDescriptor);
+		engineDescriptor.getChildren().forEach(DiscoverySelectorResolver::discoverSuite);
```

Comment:
❓ Shouldn't this be done before pruning the tree?

Reply:
Yup. :face_with_head_bandage:

Reply:
Fixed  a7bb955 .